sonar-db/src/main/java/org/sonar/db/charset/MssqlCharsetHandler.java

```diff
@@ -46,97 +42,81 @@
-    } else {
-      LOGGER.info("Verify that database columns are case-sensitive and accent-sensitive");
-    }
-  }
-
-  private void repairColumnCollation(Connection connection, ColumnDef column) throws SQLException {
+  private void repairColumnCollation(Connection connection, ColumnDef column, String expectedCollation) throws SQLException {
     // 1. select the indices defined on this column
-    String selectIndicesSql = format("SELECT I.name as index_name, I.is_unique as unik, IndexedColumns " +
-      "     FROM sys.indexes I " +
-      "     JOIN sys.tables T ON T.Object_id = I.Object_id " +
-      "     JOIN (SELECT * FROM ( " +
-      "     SELECT IC2.object_id, IC2.index_id, " +
-      "     STUFF((SELECT ' ,' + C.name " +
-      "     FROM sys.index_columns IC1 " +
-      "     JOIN sys.columns C " +
-      "     ON C.object_id = IC1.object_id " +
-      "     AND C.column_id = IC1.column_id " +
-      "     AND IC1.is_included_column = 0 " +
-      "     WHERE IC1.object_id = IC2.object_id " +
-      "     AND IC1.index_id = IC2.index_id " +
-      "     GROUP BY IC1.object_id,C.name,index_id " +
-      "     ORDER BY MAX(IC1.key_ordinal) " +
-      "     FOR XML PATH('')), 1, 2, '') IndexedColumns " +
-      "     FROM sys.index_columns IC2 " +
-      "     GROUP BY IC2.object_id ,IC2.index_id) tmp1 )tmp2 " +
-      "     ON I.object_id = tmp2.object_id AND I.Index_id = tmp2.index_id " +
-      "     WHERE I.is_primary_key = 0 AND I.is_unique_constraint = 0 " +
-      "     and T.name =('%s') " +
-      "     and CHARINDEX ('%s',IndexedColumns)>0", column.getTable(), column.getColumn());
-    List<ColumnIndex> indices = getSqlExecutor().executeSelect(connection, selectIndicesSql, ColumnIndexConverter.INSTANCE);
+    List<ColumnIndex> indices = metadata.getColumnIndices(connection, column);
 
     // 2. drop indices
     for (ColumnIndex index : indices) {
       getSqlExecutor().executeUpdate(connection, format("DROP INDEX %s.%s", column.getTable(), index.name));
     }
 
     // 3. alter collation of column
-    String csCollation = toCaseSensitive(column.getCollation());
-
     String nullability = column.isNullable() ? "NULL" : "NOT NULL";
     String size = column.getSize() >= 0 ? String.valueOf(column.getSize()) : "max";
     String alterSql = format("ALTER TABLE %s ALTER COLUMN %s %s(%s) COLLATE %s %s",
-      column.getTable(), column.getColumn(), column.getDataType(), size, csCollation, nullability);
-    LOGGER.info("Changing collation of column [{}.{}] from {} to {} | sql=", column.getTable(), column.getColumn(), column.getCollation(), csCollation, alterSql);
+      column.getTable(), column.getColumn(), column.getDataType(), size, expectedCollation, nullability);
+    LOGGER.info("Changing collation of column [{}.{}] from {} to {} | sql=", column.getTable(), column.getColumn(), column.getCollation(), expectedCollation, alterSql);
     getSqlExecutor().executeUpdate(connection, alterSql);
```

Comment:
As a general notice, DDL statements should not use a prepare statement, but a regular Statement and be executed with its `execute(String)` method.
Some DDL will fail with PreparedStatement (such as creating triggers) and some drivers will try and fail to parse the statement to find the table name (MsSql driver, iirc, seems alter is supported though) on preparedstatement. In addition, there is no benefit to use preparedStatement since it is not reused nor uses string replacements